File: example/go/config/Task/Completecondition.go
package config.Task
import (
    "config"
)
type Completecondition interface
{
    type() config.Task.Completeconditiontype

    internal virtual void _resolve(Config.LoadErrors errors)
    {
    }

    internal static Completecondition _create(Config.Stream os) {
        switch(os.ReadString()) {
            case "KillMonster":
                return config.Task.Completecondition.KillMonster._create(os);
            case "TalkNpc":
                return config.Task.Completecondition.TalkNpc._create(os);
            case "TestNoColumn":
                return config.Task.Completecondition.TestNoColumn._create(os);
            case "Chat":
                return config.Task.Completecondition.Chat._create(os);
            case "ConditionAnd":
                return config.Task.Completecondition.ConditionAnd._create(os);
            case "CollectItem":
                return config.Task.Completecondition.CollectItem._create(os);
        }
        return null;
    }
}

File: example/java/config/task/completecondition/Completecondition.java
package config.task.completecondition;

public sealed interface Completecondition permits KillMonster, TalkNpc, TestNoColumn, Chat, ConditionAnd, CollectItem {
    config.task.Completeconditiontype type();

    default void _resolve(config.ConfigMgr mgr) {
    }

    static Completecondition _create(configgen.genjava.ConfigInput input) {
        switch (input.readStr()) {
            case "KillMonster":
                return config.task.completecondition.KillMonster._create(input);
            case "TalkNpc":
                return config.task.completecondition.TalkNpc._create(input);
            case "TestNoColumn":
                return config.task.completecondition.TestNoColumn._create(input);
            case "Chat":
                return config.task.completecondition.Chat._create(input);
            case "ConditionAnd":
                return config.task.completecondition.ConditionAnd._create(input);
            case "CollectItem":
                return config.task.completecondition.CollectItem._create(input);
        }
        throw new IllegalArgumentException();
    }
}
